exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -413,4 +413,19 @@
 
   String DYNAMIC_UDF_SUPPORT_ENABLED = "exec.udf.enable_dynamic_support";
   BooleanValidator DYNAMIC_UDF_SUPPORT_ENABLED_VALIDATOR = new BooleanValidator(DYNAMIC_UDF_SUPPORT_ENABLED, true, true);
+
+  /**
+   * Option to save query profiles.
+   * <ul>
+   * <li>async (default): Write query profile after last response
+   * to the client.</li>
+   * <li>sync: Write the query profile before the last response to
+   * the client. Very useful for testing to avoid race conditions.</li>
+   * <li>none: Don't write the query profile at all. Useful when running
+   * many production jobs that do not need to be reviewed.</li>
+   * </ul>
+   */
+  String QUERY_PROFILE_OPTION = "exec.profile";
```

Comment:
exec.query_profile.write_mode might be a better choice for clarity. 
Also, would it be better to have two options ?
boolean: exec.query_profile.write_enabled and exec.query_profile.write_mode (sync or async).